docs/developers/Firmware/encryption.mdx

```diff
@@ -52,3 +52,63 @@ I'm assuming that meshtastic is being used to hike in places where someone capab
 - Two other things to keep in mind are that AES-CTR does not itself provide authenticity (e.g. an attacker can flip bits in replaying data and scramble the resulting plaintext), and that the current scheme gives some hints about transmission in the size. So, if you worry about an adversary deliberately messing-up messages or knowing the length of a text message, it looks like those might be possible.
 
 I'm guessing that the network behaves somewhat like a store-and-forward network - or, at least, that the goal is to avoid establishing a two-way connection to transmit data. I'm afraid I haven't worked with mesh networks much, but remember studying them briefly in school about ten years ago.
+
+
+## Phased Proposal for the Meshtastic Security Framework
+
+### Phase 1 - Fixed network encryption with AES-CTR
+
+The current implementation provides optional confidentiality to members of a configured network:
+ * Encryption is implemented in devices/nodes with network-wide encryption keys.
+ * Encryption is optional and is turned off when devices are in 'Ham mode'.
+ * There is no encryption supported in the clients (iOS, Android) to facilitate distribution as mass market software.
+ * Pairing from client-to-device is by:
+    * direct USB cable
+    * BT pairing
+ * Devices are 'promiscuous' and will pair with any near-by client. Network confidentiality requires physical protextion of all nodes.
+
+### Phase 2 - Strong device and client identity
+
+**Phase 2 security goals:**
+ * Know who sent a message (strong binding of messages to a particular node and/or terminal device)
```

Comment:
This would be 'data origin authentication' and 'data integrity'
Yes - message sizes will be reduced ... by 65 bytes, 64 for signature, one byte for cipher suite identifier to support later algorithm agility ... and perhaps other later optimizations (use of a keyed MAC instead of a public key signature)